protocol/triple/client.go

```diff
@@ -143,14 +143,14 @@ func newClientManager(url *common.URL) (*clientManager, error) {
 		panic(fmt.Sprintf("Unsupported serialization: %s", serialization))
 	}
 
-	// todo:// process timeout
-	// consumer config client connectTimeout
-	//connectTimeout := config.GetConsumerConfig().ConnectTimeout
+	// set timeout
+	timeout := url.GetParamDuration(constant.TimeoutKey, "")
```

Comment:
What's the default timeout value for `(constant.TimeoutKey, "")`?

Reply:
I just realized something. Are we in the middle of `invoke` here? If not, we might not be able to get the `method name` needed.

Reply:
The specific method level timeout (optional) passed in with `ctx` in `TripleInvoker.invoke(ctx)`, then to `clientManager.callUnary/callClientStream()`. If timeout presents in ctx, then it will be used as the timeout value; If not, the default value calculated here by `timeout := url.GetParamDuration(constant.TimeoutKey, "")`  will be used.

Is the above process right?

Reply:
> The specific method level timeout (optional) passed in with `ctx` in `TripleInvoker.invoke(ctx)`, then to `clientManager.callUnary/callClientStream()`. If timeout presents in ctx, then it will be used as the timeout value; If not, the default value calculated here by `timeout := url.GetParamDuration(constant.TimeoutKey, "")` will be used.
> 
> Is the above process right?

Yes, that is. The ```ctx``` used by users would have highest priority.

Reply:
> I just realized something. Are we in the middle of an `RPC invoke` here? If not, we might not be able to get the `method name` needed.

We are in the process of ```Refer```, i.e. initialize client and some general configurations.

Reply:
> > The specific method level timeout (optional) passed in with `ctx` in `TripleInvoker.invoke(ctx)`, then to `clientManager.callUnary/callClientStream()`. If timeout presents in ctx, then it will be used as the timeout value; If not, the default value calculated here by `timeout := url.GetParamDuration(constant.TimeoutKey, "")` will be used.
> > Is the above process right?
> 
> Yes, that is. The `ctx` used by users would have highest priority. It is more ideal for uses to use `ctx` to specify timeout for method level since they can use `cancel` function directly.

I see, this sounds reasonable to me.

One more thing, we need to check the `method level timeout` if `ctx` is not set, then put `method timeout` into `ctx`.